analyzers/tests/SonarAnalyzer.CSharp.Styling.Test/TestCases/UseNullInsteadOfDefault.cs

```diff
@@ -0,0 +1,181 @@
+﻿using System;
+using System.Collections.Generic;
+
+class UseNullInsteadOfDefault
+{
+    void Method()
+    {
+        int integer = default;
+        int x = default, y = 0, z = default(int);
+        var integer2 = default(Int32);
+        object objectReference = default; // Noncompliant {{Use 'null' instead of 'default' for reference types.}}
+        //                       ^^^^^^^
+        var objectReference2 = default(object); // Noncompliant {{Use 'null' instead of 'default' for reference types.}}
+        //                     ^^^^^^^^^^^^^^^
+        object obj1 = default, obj2 = null, obj3 = default(object);
+        //            ^^^^^^^
+        //                                         ^^^^^^^^^^^^^^^@-1
+        IEnumerable<string> collection = null;
+        IEnumerable<string> collection2 = default; // Noncompliant
+        IEnumerable<string> collection3 = true ? null : default; // Noncompliant
+        collection = null;
+        collection2 = default; // Noncompliant
+        collection2 = true ? null : default; // Noncompliant
+
+        _ = default(object) is null; // Noncompliant FP - Do we care?
+        _ = collection is default(IEnumerable<string>); // Noncompliant
+
+
+        int? nullableInt = default; // Noncompliant
+
+        _ = collection == default; // Noncompliant
+
+        switch (integer)
+        {
+            case default(int):
+                break;
+            default:
+                break;
+        }
+
+        switch (collection)
+        {
+            case default(IEnumerable<string>): // Noncompliant
+                break;
+            default:
+                break;
+        }
+    }
+
+    void GenericMethod<T>()
+    {
+        T t = default;
+        var t2 = default(T);
+        T? t3 = default;
+        T? t4 = default(T);
+
+        t = default;
+        t = default(T);
+        t3 = default;
+        t3 = default(T);
+
+        _ = t == default; // Error [CS8761]
+        // Error@+1 [CS8761]
+        _ = t3 == default;
+        _ = t is default(T); // Error [CS0150]
+        // Error@+1 [CS0150]
+        _ = t3 is default(T);
+
+        switch (t)
+        {
+            case default(T): // Error [CS0150]
+                break;
+            default:
+                break;
+        }
+
+        switch (t3)
+        {
+            // Error@+1 [CS0150]
+            case default(T):
+                break;
+            default:
+                break;
+        }
+    }
+
+    void GenericMethodClassConstraint<T>()
+        where T : class
+    {
+        T t = default; // Noncompliant
+        var t2 = default(T); // Noncompliant
+        T? t3 = default; // Noncompliant
+        T? t4 = default(T); // Noncompliant
```

Comment:
They are the same, would you expect to read `default(T)`?

Reply:
The exercise should demonstrate that
```
// Noncompliant {{Use 'null' instead of 'default' for reference types.}}
// Noncompliant {{Use 'null' instead of 'default' for reference types.}}
// Noncompliant {{Use 'null' instead of 'default' for reference types.}}
// Noncompliant {{Use 'null' instead of 'default' for reference types.}}
```
is trivial to review, while
```
// Noncompliant {{Use 'null' instead of 'default' for reference types.}}
   // Noncompliant {{Use 'null' instead of 'default' for reference types.}}
 // Noncompliant {{Use 'null' instead of 'default' for reference types.}}
  // Noncompliant {{Use 'null' instead of 'default' for reference types.}}
```
is impossible to review.

The code itself is fine.

Reply:
You can resolve this, I just want to make sure you got the reply